Create Approvals test doc (#6821)
thepracticaldev/dev.to: docs/tests/approvals-tests.md

---
title: Approvals Tests
---

# Approvals Tests

> Approvals are based on the idea of the golden master. You take a snapshot of an

Reviewer: I think a link to a something that explains the Golden Master will be very beneficial here. I don't think everyone will know what it is.

Author: I think I'm going to go ahead and remove that as it's not really necessary.

> Approvals are based on the idea of the golden master. You take a snapshot of an

Reviewer: I feel like the sentence 'You take a snapshot' is not very articulate here. I would be more specific if possible and replace you with specificity.

Author: I quite literally took this from their [official repo](https://github.com/kytrinyx/approvals). Yeah, I agree that pluralizing it sounds a bit off but I'm just keeping it consistent with the gem name.

object and then compare all future versions of the object to the snapshot. A

> good use-case for Approvals tests is comparing rendered HTML.

Reviewer: ```suggestion
good use-case for the Approvals tests is to compare the rendered HTML to the test specified HTML.
```

I feel like when using the word compared you usually need to specify what to its being compared to.

## Usage

> Simply create your test similar to the following

Reviewer: ```suggestion
Simply create your test similar to the following:
```

```ruby

> it "works" do

Reviewer: ```suggestion
it "renders the correct HTML on the home page" do
```
I feel like a more realistic test description will be more beneficial

  page = '<html><head></head><body><h1>ZOMG</h1></body></html>'
  verify(format: :html) { page } # format can also be :json
end
```

> then run the said test and a new `*.received.*` file will be created. You will

Reviewer: ```suggestion
You may then run the said test and a new `*.received.*` file will be created. Thereafter,
```

> then run the following to verify the newly created file and approve the created

Reviewer: ```suggestion
run the following command to verify the newly created file and approve the created
```

change.

```shell
approvals verify
```

Please be sure to include the Approvals file in your commit.

## Edge cases

> Approvals test isn't great for testing changing variables. That includes

Reviewer: ```suggestion
Approvals tests are difficult to utilise for testing variables that change. This includes variables like:
```
